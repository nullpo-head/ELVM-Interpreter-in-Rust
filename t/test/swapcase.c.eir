	.text
main:
	mov D, SP
	add D, -1
	store BP, D
	mov SP, D
	mov BP, SP
	sub SP, 1
	.file 1 "out/swapcase.c"
	.loc 1 12 0
	# }
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov A, 0
	mov B, SP
	getc A
	jne .L0, A, 0
	mov A, -1
	.L0:
	mov B, BP
	add B, 16777215
	store A, B
	.loc 1 9 0
	#   }
	.loc 1 6 0
	#     c += 32;
	mov A, 65
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov B, A
	load A, SP
	add SP, 1
	le A, B
	mov B, 0
	jeq .L1, A, 0
	.loc 1 6 0
	#     c += 32;
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	.loc 1 6 0
	#     c += 32;
	mov A, 90
	mov B, A
	load A, SP
	add SP, 1
	le A, B
	mov B, A
	ne B, 0
	.L1:
	mov A, B
	jeq .L2, A, 0
	.loc 1 7 0
	#   } else if (c >= 'a' && c <= 'z') {
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	.loc 1 7 0
	#   } else if (c >= 'a' && c <= 'z') {
	mov A, 32
	mov B, A
	load A, SP
	add SP, 1
	add A, B
	mov B, BP
	add B, 16777215
	store A, B
	jmp .L3
	.L2:
	.loc 1 9 0
	#   }
	.loc 1 7 0
	#   } else if (c >= 'a' && c <= 'z') {
	mov A, 97
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov B, A
	load A, SP
	add SP, 1
	le A, B
	mov B, 0
	jeq .L4, A, 0
	.loc 1 7 0
	#   } else if (c >= 'a' && c <= 'z') {
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	.loc 1 7 0
	#   } else if (c >= 'a' && c <= 'z') {
	mov A, 122
	mov B, A
	load A, SP
	add SP, 1
	le A, B
	mov B, A
	ne B, 0
	.L4:
	mov A, B
	jeq .L5, A, 0
	.loc 1 9 0
	#   }
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	.loc 1 9 0
	#   }
	mov A, 32
	mov B, A
	load A, SP
	add SP, 1
	sub A, B
	mov B, BP
	add B, 16777215
	store A, B
	.L5:
	.L3:
	.loc 1 11 0
	#   return 0;
	.loc 1 5 0
	#   if (c >= 'A' && c <= 'Z') {
	mov B, BP
	add B, 16777215
	load A, B
	mov D, SP
	add D, -1
	store A, D
	mov SP, D
	putc A
	add SP, 1
	.loc 1 12 0
	# }
	mov A, 0
	mov B, A
	exit
	exit
