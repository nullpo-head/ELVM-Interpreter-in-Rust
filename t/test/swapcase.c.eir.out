=== test/swapcase_1.in ===
0
=== test/swapcase_2.in ===
a
=== test/swapcase_3.in ===
z
=== test/swapcase_4.in ===
A
=== test/swapcase_5.in ===
Z
=== test/swapcase_6.in ===
$
